impress-org/givewp
# Fix: check for test mode to avoid PayPal IPN validation error

## Description

PayPal Standard Gateway > IPN listener > HTTP request "Host" header should match request host.
https://www.paypal.com/dm/smarthelp/article/how-do-i-modify-my-ipn-php-listener-to-support-http1.1-ts1492
https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.23
When GiveWP in test mode the "Host" header is hardcoded to 'www.paypal.com' creating an "IPN Error". (See Visuals).
PR fixes the bug by checking if GiveWP is in test mode, and thus setting "Host" header to 'www.sandbox.paypal.com'

## Affects

PayPal Standard Gateway

## Visuals
![CleanShot 2021-10-25 at 23 32 50](https://user-images.githubusercontent.com/18058037/138767392-c0e77cf8-f322-456d-a0ec-08a086e90ad0.jpeg)

## Testing Instructions

1. Enable GiveWP Test Mode
2. Use PayPal Standard Gateway
3. Make a payment 
4. Look at the logs wp-admin/edit.php?post_type=give_forms&page=give-tools&tab=logs to see "IPN Error"

## Pre-review Checklist
